rollouts/controllers/rollout_controller.go

```diff
@@ -451,18 +443,36 @@ func (r *RolloutReconciler) computeTargets(ctx context.Context,
 	return targets, nil
 }
 
-func pkgNeedsUpdate(rollout *gitopsv1alpha1.Rollout, rrs gitopsv1alpha1.RemoteRootSync, pkg *packagediscovery.DiscoveredPackage) (*gitopsv1alpha1.RemoteRootSync, bool) {
+func pkgNeedsUpdate(rollout *gitopsv1alpha1.Rollout, rrs gitopsv1alpha1.RemoteSync, pkg *packagediscovery.DiscoveredPackage) (*gitopsv1alpha1.RemoteSync, bool) {
 	// TODO: We need to check other things here besides git.Revision and metadata
-	if pkg.Revision != rrs.Spec.Template.Spec.Git.Revision ||
-		!reflect.DeepEqual(rollout.Spec.SyncTemplate.RootSync.Metadata,
-			rrs.Spec.Template.Metadata) {
-		rrs.Spec.Template.Spec.Git.Revision = pkg.Revision
-		rrs.Spec.Template.Metadata = rollout.Spec.SyncTemplate.RootSync.Metadata
+	metadata := getSpecMetadata(rollout)
```

Comment:
You may want to get this change sooner since results in panic for existing rollout objects :)